js/usernameSelection.js

```diff
@@ -72,57 +73,52 @@ const removeUsername = function removeUsername() {
-    showingAllUsernames = false;
-    renderUsernames();
-};
-
-const renderShowMoreLessUsernames = function renderShowMoreLessUsernames() {
-    const moreLessUsernamesElement = document.getElementById('showMoreLessUsernames');
-    if (showingAllUsernames) { moreLessUsernamesElement.innerHTML = '<a href=javascript:showLessUsernames()><strong>Show less usernames</strong></a>'; } else { moreLessUsernamesElement.innerHTML = '<a href=javascript:showAllUsernames()><strong>Show more usernames</strong></a>'; }
+    console.log("showUsernames clicked");
+    var usernameSelector = document.getElementById("username_selector");            
+    usernameSelector.style.display = usernameSelector.style.display == "none" ? "block" : "none";
 };
 
 const generateUsernameSelector = function generateUsernameSelector() {
     let usernameSelector = '';
     let i = 0;
-    if (showingAllUsernames || USERNAMES.length <= MAX_USERNAMES_SHOWING) {
-        for (; i < USERNAMES.length - 1; i += 1) {
-            usernameSelector += `<a class='selectors' href='https://github.com/${USERNAMES[i]}?tab=stars'>${USERNAMES[i]}</a>`;
-            usernameSelector += ' | ';
-        }
-    } else {
-        for (; i < MAX_USERNAMES_SHOWING - 1; i += 1) {
-            usernameSelector += `<a class='selectors' href='https://github.com/${USERNAMES[i]}?tab=stars'>${USERNAMES[i]}</a>`;
-            usernameSelector += ' | ';
-        }
-    }
-    usernameSelector += `<a class='selectors' href='https://github.com/${USERNAMES[i]}?tab=stars'>${USERNAMES[i]}</a>`;
+    let usernames = JSON.parse(localStorage.getItem('usernames'));
+    for (; i < usernames.length - 1; i += 1) {
+        usernameSelector += `<a class='selectors' href='https://github.com/${usernames[i]}?tab=stars'>${usernames[i]}</a>`;
+        usernameSelector += ' | ';
+    }        
+    usernameSelector += `<a class='selectors' href='https://github.com/${usernames[i]}?tab=stars'>${usernames[i]}</a>`;
     return usernameSelector;
 };
 
 const renderUsernames = function renderUsernames() {
-    document.getElementById('username_selector').innerHTML = generateUsernameSelector();
-    document.getElementById('addOrRemoveUsername').innerHTML = "<a class='addOrRemoveUsername' href=javascript:addUsername()><strong>Add more username</strong></a>" +
-        " | <a class='addOrRemoveUsername' href=javascript:removeUsername()><strong>Remove username</strong></a>";
-    renderShowMoreLessUsernames();
-};
\ No newline at end of file
+    document.getElementById("username_selector").innerHTML = generateUsernameSelector();
+    document.getElementById("addUsername").setAttribute("href", "javascript:addUsername()");     
+    document.getElementById("removeUsername").setAttribute("href", "javascript:removeUsername()"); 
+    document.getElementById("showUsernames").setAttribute("href", "javascript:showAllUsernames()"); 
+
+    document.getElementById("settings").onclick = function() {
```

Comment:
I'm a beginner with Javascript. Should I change it to something like this?

~~~
function renderUsernames() {

document.getElementById("settings").addEventListener('click', callback);

function callback() {

        var settings = document.getElementById("dropdown_content");

        console.log("settings clicked and " + settings.style.display);

        settings.style.display = settings.style.display == (null || "none") ? "block" : "none";

    };

};
~~~